### ISSUE TITLE
Run configuration for testing: cannot find shared objects/dlls

### ISSUE THREAD

**Issue by JoostvanPinxten** [OneMoreGres/qtc-gtest#10]
I've got some shared objects that are picked up when launching the full test suite, but when run through the menu (say: Tools -> Google Test -> Check Project)

```Starting Debug/tests/tests...
Debug/tests/tests: error while loading shared libraries: libcommon.so.1: cannot open shared object file: No such file or directory
Debug/tests/tests exited with code 127```

I can work around it by launching it the normal way, but then I can't use the awesome feature that only a part of my test set is being run.

I'm assuming that some part of the run configuration isn't setup completely, or not completely copied from the others.

BTW: On the whole, it's working great under both Windows and Linux; I also like the easy installation :-)

**Comment by OneMoreGres** [comment]
Ok, i'll look for it. Some questions for now:
* does "Debug/tests/tests" really exists or it should be just "Debug/tests"?
* is libcommon.so placed in system environment (PATH, /usr/lib, etc) or you add it by modifying build/run environment in Qt Creator? or even add it via command args (LD_LIBRARY_PATH=...) ?
* what is a type of your app: qmake/cmake/qbs/remote/...?

**Comment by JoostvanPinxten** [comment]
Thanks. The project in question is my [TSPLIB parser](https://github.com/JoostvanPinxten/tsplib-parser). It doesn't quite function as minimal example (due to the dependencies to Flex and Bison), but I can at least explain (and show) you the project structure. (So yes, Debug/tests/test/ really exists, it's just truncated from the complete path).

Anyhow, I have a local `qmake` project that contains a few `subdir` projects. Those subdirs create either shared libs (`.so` or `.dll`) or they create a main executable by linking to those. I refer only to `pthreads` as an external library, the rest are internal. See e.g. [the tests project file](https://github.com/JoostvanPinxten/tsplib-parser/blob/master/tests/tests.pro).

Gtest itself is also one of those libraries, built in the same project. Now, I can run them like you would normally launch an app, but the run configurations from the menu do not work (haven't tried it on Windows yet though). I do not have to modify the library paths, Qt Creator normally takes care of that. When I run it from the command-line, I do need to point to those libs through `LD_LIBRARY_PATH`.

I hope that answers most of your questions!

**Comment by OneMoreGres** [comment]
Fixed. I'll recompile and publish them with QTC 3.5.0 release. For now you can download and test it from here:
http://rghost.ru/download/private/6bQ64JL95/d08e5d3287aa2e28abaeb80e29f9799b/ca405085f8d4d8060f7eeae5d3030b4ac422a6e0/QtcGtest-3.4.0-linux-x64.tar.gz
http://rghost.ru/download/private/6t8Lp6mdJ/20de261b753009910168befcc1070cba/219044aed3825fc54f9f3b9bfce32042193138bb/QtcGtest-3.4.0-win.zip

**Comment by JoostvanPinxten** [comment]
Under Ubuntu I can now indeed run the project through the menu item (Tools -> Google Test -> Check project).

However, I received a "Program has terminated unexpectedly" under Windows. Is it possible that it's not quite complete yet?

Also, when running the other two options (Check changed, Check current), I do a proper run, but the filters don't seem to be set properly:

```Starting (omitted)/parser/tests/tests...
Running main() from gtest_main.cc
Note: Google Test filter = SIMPLE_CULL.*:*/SimpleCullTest.*:*/SimpleCullTest_dominated.*
[==========] Running 0 tests from 0 test cases.
[==========] 0 tests from 0 test cases ran. (6 ms total)
[  PASSED  ] 0 tests.
(omitted)/parser/tests/tests exited with code 0```

BTW: This test contains 2 parameterized tests (SimpleCullTest and SimpleCullTest_dominated) and a few 'normal' tests (under the SIMPLE_CULL case).

And thanks for the feedback! I'll look into it, I'm not completely up to speed with the directives of qmake... I noticed that the development version hadn't been completely pushed yet... I will fix these small issues soon.

**Comment by OneMoreGres** [comment]
QTC crashed under Windows? Hmm.. Maybe i've compiled it against wrong qtc or MSVC version... I'll check it. And filters too. But not right now :)

**Comment by OneMoreGres** [comment]
Fixed again. Now i've tested it and on windows too.
http://rghost.ru/download/private/7B6fvp4KW/9e9f046e6d56e56a7715ec04363e60fb/24c9b77f5bf17d39ce46445bb2d8c279b2857b4e/QtcGtest-3.4.0-linux-x64.tar.gz
http://rghost.ru/download/private/8MsHD7Xwb/6613f61c1668f014cad4495fd944b1e7/3fa097902575ec6cf432e9590c2a25f5ddc0a859/QtcGtest-3.4.0-win.zip

  > md5sum QtcGtest-3.4.0-linux-x64.tar.gz 
  8985ca9c131c4781e742ea9c12f4dba7  QtcGtest-3.4.0-linux-x64.tar.gz

  > md5sum QtcGtest-3.4.0-win.zip 
  ad4e86e99335eb8e9eebbdc1185dbe54  QtcGtest-3.4.0-win.zip

---
About tests. There are no SimpleCull* tests in repository. Possible it is one of those unpushed changes you mentioned. Tell me after you push them and i'll check filter.

**Comment by JoostvanPinxten** [comment]
I noticed that I had selected the 'wrong' tests-executable (I have two in my main application, one for the application, the other one inherited from tsplibparser). The tests indeed do not exist in that executable. Not completely a fool-proof approach; I think it is difficult to select the proper project/executable from the filter? Although the current file might give a hint to which executable is actually supposed to be run, that doesn't guarantee it either...
